tap_mailchimp/sync.py

```diff
@@ -302,10 +304,13 @@ def transform_activities(records):
 
 def sync_email_activity(client, catalog, state, start_date, campaign_ids, batch_id=None):
     if batch_id:
-        LOGGER.info('reports_email_activity - Picking up previous run: {}'.format(batch_id))
+        LOGGER.info('reports_email_activity - Picking up previous run: %s', batch_id)
     else:
         LOGGER.info('reports_email_activity - Starting sync')
 
+        selected_fields = get_selected_fields(catalog, 'reports_email_activity')
+        field_names = ['emails.'+str(x) for x in selected_fields]+['_links', 'total_items', 'constraints','emails.'+str('_links')]
```

Comment:
This could be moved into `get_selected_fields` and be clarified with its intent. Like adding the "top level fields"